src/app/store/slices/plan/index.ts

```diff
@@ -1,152 +1,141 @@
   individualPlan: StoragePlan | null;
+  businessPlan: StoragePlan | null;
   teamPlan: StoragePlan | null;
   planLimit: number;
   planUsage: number;
   usageDetails: UsageResponse | null;
-  subscription: UserSubscription | null;
-}
-
-interface FetchPlansResult {
-  individualPlan: StoragePlan | null;
-  teamPlan: StoragePlan | null;
+  individualSubscription: UserSubscription | null;
+  businessSubscription: UserSubscription | null;
+  businessPlanLimit: number;
+  businessPlanUsage: number;
+  businessPlanUsageDetails: UsageResponse | null;
 }
 
 const initialState: PlanState = {
   isLoadingPlans: false,
   isLoadingPlanLimit: false,
   isLoadingPlanUsage: false,
   individualPlan: null,
+  businessPlan: null,
   teamPlan: null,
   planLimit: 0,
   planUsage: 0,
   usageDetails: null,
-  subscription: null,
+  individualSubscription: null,
+  businessSubscription: null,
+  businessPlanLimit: 0,
+  businessPlanUsage: 0,
+  businessPlanUsageDetails: null,
 };
 
 export const initializeThunk = createAsyncThunk<void, void, { state: RootState }>(
   'plan/initialize',
   async (payload: void, { dispatch, getState }) => {
     const isAuthenticated = getState().user.isAuthenticated;
     const promises: Promise<void>[] = [];
+    const { workspaces, selectedWorkspace } = getState().workspaces;
+    const workspaceUserId = selectedWorkspace?.workspace?.workspaceUserId || workspaces[0]?.workspace?.workspaceUserId;
 
     if (isAuthenticated) {
-      promises.push(dispatch(fetchPlans()).then());
+      promises.push(dispatch(fetchSubscriptionThunk({ userType: UserType.Individual })).then());
+      promises.push(dispatch(fetchSubscriptionThunk({ userType: UserType.Business })).then());
       promises.push(dispatch(fetchLimitThunk()).then());
       promises.push(dispatch(fetchUsageThunk()).then());
-      promises.push(dispatch(fetchSubscriptionThunk()).then());
+      if (workspaceUserId) {
+        promises.push(dispatch(fetchLimitThunk(workspaceUserId)).then());
+        promises.push(dispatch(fetchUsageThunk(workspaceUserId)).then());
+      }
     }
 
     await Promise.all(promises);
   },
 );
 
-export const fetchPlans = createAsyncThunk<FetchPlansResult, void, { state: RootState }>(
-  'plan/fetchPlans',
-  async (payload: void, { getState }) => {
-    const user = getState().user.user;
-    const promises: Promise<StoragePlan | null>[] = [];
-
-    promises.push(planService.fetchIndividualPlan());
-    if (user?.teams) {
-      promises.push(planService.fetchTeamPlan());
-    }
-
-    const [individualPlan, teamPlan] = await Promise.all(promises);
-
-    return { individualPlan, teamPlan };
-  },
-);
-
-export const fetchLimitThunk = createAsyncThunk<number, void, { state: RootState }>(
+export const fetchLimitThunk = createAsyncThunk<number, string | undefined, { state: RootState }>(
   'plan/fetchLimit',
-  async (payload: void, { getState }) => {
+  async (payload: string | undefined, { getState }) => {
```

Comment:
what is the payload? Add an object to clarify with the name of the fields needed by the payload